pallets/moonbeam-lazy-migrations/src/lib.rs

```diff
@@ -106,5 +127,61 @@ pub mod pallet {
 			)
 			.into())
 		}
+
+		// TODO(rodrigo): This extrinsic should be removed once the storage of destroyed contracts
+		// has been removed
+		#[pallet::call_index(1)]
+		#[pallet::weight({
+			let addresses_len = addresses.len() as u32;
+			<T as crate::Config>::WeightInfo::clear_suicided_storage(addresses_len, *limit)
+		})]
+		pub fn clear_suicided_storage(
+			origin: OriginFor<T>,
+			addresses: BoundedVec<H160, GetArrayLimit>,
+			limit: u32,
+		) -> DispatchResultWithPostInfo {
+			ensure_signed(origin)?;
+
+			ensure!(limit != 0, Error::<T>::LimitCannotBeZero);
+			ensure!(
+				addresses.len() != 0,
+				Error::<T>::AddressesLengthCannotBeZero
+			);
+
+			let mut limit = limit as usize;
+
+			for address in &addresses {
+				// Ensure that the contract is corrupted by checking
+				// that it has no code and at least one storage entry.
+				let suicided = pallet_evm::Suicided::<T>::contains_key(&address);
+				let has_code = pallet_evm::AccountCodes::<T>::contains_key(&address);
+				ensure!(
+					!suicided
+						&& !has_code && pallet_evm::AccountStorages::<T>::iter_key_prefix(&address)
+						.next()
+						.is_some(),
+					Error::<T>::ContractNotCorrupted
+				);
+
+				let deleted = pallet_evm::AccountStorages::<T>::drain_prefix(*address)
+					.take(limit)
+					.count();
+
+				// Check if the storage of this contract has been completly removed
+				if pallet_evm::AccountStorages::<T>::iter_key_prefix(&address)
```

Comment:
What happens if there are still entries? should we call this again? how do we know there are still pending items? I wonder if it is better to emit events vs keeping a count in storage?

Reply:
This will be called multiple times until all the storage entries are removed. If a single address consumes the limit we provide, we just have to call it again for the same address. This is required because a contract can contain a big chunk of storage which cannot be removed in a single block.

Reply:
But how does the caller know what address to use? say I send 500 addresses and the limit is reached at address 234. Should the caller manually check the diff in `SuicidedContractsRemoved`?

Reply:
The call only ensures the only corrupted contracts will be removed and also that how tries to call it respects the pov limits.